hypnotoad/core/equilibrium.py

```diff
@@ -2907,13 +2912,12 @@ def __init__(self, nonorthogonal_settings):
             nonorthogonal_settings
         )
 
-    def updateOptions(self):
-        """
-        Set default values from user_options
-        """
-        if hasattr(self, "regions"):
-            for region in self.regions.values():
-                region.setupSpacing()
+    def resetNonorthogonalOptions(self, nonorthogonal_settings):
```

Comment:
I'd prefer to create a new object if possible, rather than reset the options, in case this object is shared elsewhere. Is modifying options definitely needed?

Reply:
I think it is necessary - once I update #26 you'll be able to try this out, but when creating nonorthogonal grids, the initial construction (creating `FineContour`s and finding wall intersections) takes a long time. Then often the grids look weird for one reason or another, and the way to fix that is to adjust the poloidal spacing of the grid points, which can be done much more quickly than building the `Mesh` object in the first place, because the `FineContour`s don't need to change, so all the work done in constructing them can just be reused. It does make the workflow a lot better, so I think it's worth the compromise of allowing a small number of options to be changed. I've tried to make sure that there is no persistent state in the `Mesh`/`Equilibrium`/`MeshRegion`/`EquilibriumRegion` that depends on `nonorthogonal_options` except for the actual position of the grid points to minimise the chances of some odd history-dependant thing happening. Would still recommend building the grid once from the ground up with the final settings once they're decided on to be safe (and maybe we should enforce that in the GUI at some point).